include/avk/cpp_utils.hpp

```diff
@@ -1222,24 +1222,35 @@ namespace avk
 	{
 		if (!aOld.has_value()) {
 			aOld = std::move(aNew);
-		} 
+		}
 		else {
 			std::swap(aNew, aOld);
 			aLifeTimeHandler(std::move(aNew));
 		}
 	}
-		
+
 	template<typename T, typename F> requires boolean_convertible<T>
 	void emplace_and_handle_previous(T& aNew, T&& aOld, F&& aLifeTimeHandler)

```

Comment:
Please comment this function! (Especially the fact that `aOld` will have its guts stolen).

Reply:
added more precise comments for all emplace_and_handle_previous definitions.